### ISSUE TITLE
[Bug]: `aws_elasticache_replication_group` transit_encryption_enable changes cause resource replacement

### ISSUE THREAD

**Issue by jkoermer-eqxm** [hashicorp/terraform-provider-aws#30700]
### Terraform Core Version

0.12.31

### AWS Provider Version

4.62

### Affected Resource(s)

aws_elasticache_replication_group

### Expected Behavior

Updating the seeing for transit encryption enabled:
transit_encryption_enabled = true | false

Should just modify the cluster.  Running any either of the following commands will modify the existing cluster and not require the cluster to be rebuilt:
aws elasticache modify-replication-group --replication-group-id cache-cluster --transit-encryption-enabled --transit-encryption-mode preferred --apply-immediately
aws elasticache modify-replication-group --replication-group-id cache-cluster--no-transit-encryption-enabled --apply-immediately

### Actual Behavior

Running terraform plan or apply cause the resource to be recreated:
` ~ transit_encryption_enabled     = true -> false # forces replacement`

Modifying the transit encryption outside of terraform and updating terraform to match allows for the resource to plan/apply as expected. 

### Terraform Configuration Files

```
resource "aws_elasticache_replication_group" "redis" {
  count                         = var.clustered_cache ? 1 : 0

  lifecycle {
    # Comment out ignore_changes to force upgrade based on version change
    prevent_destroy = false
    ignore_changes = [engine_version]
  }

  automatic_failover_enabled    = local.automatic_failover_enabled
  multi_az_enabled              = local.multi_az_enabled
  replication_group_id          = "${var.owner}-${var.application}-${var.environment}${var.suffix}"
  description                   = "MultiAZ ${var.owner}-${var.application}-${var.environment}${var.suffix} Cluster"
  port                          = var.elasticache_port
  parameter_group_name          = aws_elasticache_parameter_group.parameter_group_redis.id

  node_type                     = var.node_type
  num_cache_clusters            = var.node_count
  num_node_groups               = var.num_node_groups
  replicas_per_node_group       = var.replicas_per_node_group

  snapshot_name                 = var.snapshot_name
  snapshot_retention_limit      = local.backup_days
  snapshot_window               = var.backup_window
  maintenance_window            = var.maintenance_window
  auto_minor_version_upgrade    = var.auto_minor_version_upgrade
  engine                        = "redis"
  engine_version                = var.engine_version
  security_group_ids            = [aws_security_group.redis.id]
  subnet_group_name             = local.subnet_group_name
  apply_immediately             = var.apply_immediately

  at_rest_encryption_enabled    = var.at_rest_encryption_enabled
  auth_token                    = var.auth_token      # This is required if transit_encryption_enabled is set
  kms_key_id                    = var.at_rest_encryption_enabled == true ? local.kms_key_id : null
  
  transit_encryption_enabled    = var.transit_encryption_enabled # MODIFY THIS VALUE 
  tags                     = local.tags
}
```

### Steps to Reproduce

Create an elasticache instance.
Modify the value of `transit_encryption_enabled`

### References

This is related to a couple of additional open issues relating to elasticache transit encryption.  Including:
https://github.com/hashicorp/terraform-provider-aws/issues/29403
https://github.com/hashicorp/terraform-provider-aws/issues/26367
https://docs.aws.amazon.com/cli/latest/reference/elasticache/modify-cache-cluster.html

I think this gets resolved with this PR:
https://github.com/hashicorp/terraform-provider-aws/pull/30403

### Would you like to implement a fix?

No

**Comment by teddy-wahle** [comment]
[AWS docs ](https://docs.aws.amazon.com/AmazonElastiCache/latest/mem-ug/in-transit-encryption-mc.html)say, "You can enable in-transit encryption on a cluster only when creating the cluster. You cannot toggle in-transit encryption on and off by modifying a cluster." which indicates to me that this is not a fixable or a bug.

**Comment by stefansundin** [comment]
I didn't post a comment here but a few months ago I created this PR which fixes this issue: https://github.com/hashicorp/terraform-provider-aws/pull/30403

Add an upvote on the PR and hopefully hashicorp takes a look at it more quickly. :)

(GitHub automatically posts an "activity" item above but it isn't a comment so it is easy to miss.)

**Comment by xposix** [comment]
This was fixed in 5.16.0 💛
